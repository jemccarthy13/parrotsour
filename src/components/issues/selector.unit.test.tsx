import React from "react"
import { act, fireEvent, render } from "@testing-library/react"
import { userEvent } from "@testing-library/user-event"
import { vi } from "vitest"
import IssueSelector from "./selector"

const mockChangeFn = vi.fn(() => {
  return vi.fn()
})

describe("IssueSelector_Component", () => {
  afterEach(() => {
    mockChangeFn.mockClear()
  })

  const featureSelector = /feature/i
  const picSelector = /this picture/i
  const othSelector = /Other/i

  it("renders_picprob_default_checked", () => {
    const selWrapper = render(<IssueSelector onChange={mockChangeFn} />)

    expect(
      (selWrapper.getByLabelText(picSelector) as HTMLInputElement).checked
    ).toEqual(true)
    expect(selWrapper).toMatchSnapshot()
  })

  it.skip("alerts_parent_only_when_sel_changes", async () => {
    userEvent.setup()

    const selWrapper = render(<IssueSelector onChange={mockChangeFn} />)

    act(() => {
      selWrapper.getByLabelText(othSelector).focus()
    })

    act(() => {
      userEvent.click(selWrapper.getByLabelText(othSelector))
    })

    expect(mockChangeFn).toHaveBeenCalledTimes(1)
    expect(mockChangeFn).toHaveBeenCalledWith("othprob")

    act(() => {
      fireEvent.click(selWrapper.getByLabelText(featureSelector))
    })

    expect(mockChangeFn).toHaveBeenCalledTimes(2)
    expect(mockChangeFn).toHaveBeenCalledWith("feature")
  })
})
