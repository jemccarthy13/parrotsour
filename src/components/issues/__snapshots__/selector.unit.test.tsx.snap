// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IssueSelector_Component > renders_picprob_default_checked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          <u>
            Problem:
          </u>
        </h2>
        <div
          style="margin-bottom: 24px;"
        >
          See a list of
           
          <a
            href="/#/changelog"
            style="color: rgb(66, 165, 245);"
          >
            known issues
          </a>
          .
        </div>
        <div
          aria-labelledby="demo-controlled-radio-buttons-group"
          class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-j8yymo"
                name="controlled-radio-buttons-group"
                type="radio"
                value="picprob"
              />
              <span
                class="css-1qiat4j"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z8nmqa-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
            >
              The answer to this picture is incorrect
              <div
                class="css-1k0kukj"
              >
                A copy of the image is submitted with this report
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-j8yymo"
                name="controlled-radio-buttons-group"
                type="radio"
                value="feature"
              />
              <span
                class="css-1qiat4j"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
            >
              Feature request
              <div
                class="css-1k0kukj"
              >
                Wouldnâ€˜t it be nice if...
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-j8yymo"
                name="controlled-radio-buttons-group"
                type="radio"
                value="othprob"
              />
              <span
                class="css-1qiat4j"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
            >
              Other
            </span>
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        <u>
          Problem:
        </u>
      </h2>
      <div
        style="margin-bottom: 24px;"
      >
        See a list of
         
        <a
          href="/#/changelog"
          style="color: rgb(66, 165, 245);"
        >
          known issues
        </a>
        .
      </div>
      <div
        aria-labelledby="demo-controlled-radio-buttons-group"
        class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
        role="radiogroup"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-j8yymo"
              name="controlled-radio-buttons-group"
              type="radio"
              value="picprob"
            />
            <span
              class="css-1qiat4j"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z8nmqa-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
          >
            The answer to this picture is incorrect
            <div
              class="css-1k0kukj"
            >
              A copy of the image is submitted with this report
            </div>
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
          >
            <input
              class="PrivateSwitchBase-input css-j8yymo"
              name="controlled-radio-buttons-group"
              type="radio"
              value="feature"
            />
            <span
              class="css-1qiat4j"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
          >
            Feature request
            <div
              class="css-1k0kukj"
            >
              Wouldnâ€˜t it be nice if...
            </div>
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
          >
            <input
              class="PrivateSwitchBase-input css-j8yymo"
              name="controlled-radio-buttons-group"
              type="radio"
              value="othprob"
            />
            <span
              class="css-1qiat4j"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
          >
            Other
          </span>
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
